doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+
+Getting started with config.utils.schema
+----------------------------------------
+
+As an example, consider an :ref:`application role <configuration_application_roles>`
+that has a single configuration option - an HTTP endpoint address.
+
+..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_scalar/config.yaml
+    :language: yaml
+    :start-at: roles
+    :dedent:
+
+This is how you can use the ``experimental.config.utils.schema`` module to process
+the role configuration:
+
+1.  Load the module:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_scalar/http_api.lua
+        :language: lua
+        :start-at: config.utils.schema
+        :end-at: config.utils.schema
+        :dedent:
+
+2.  Define a *schema* -- the root object that stores information about the role's
+    configuration -- using :ref:`schema.new() <config-utils-schema-new>`. The example
+    below shows a schema that includes a single string option:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_scalar/http_api.lua
+        :language: lua
+        :start-at: local http_api_schema
+        :end-before: local function validate
+        :dedent:
+
+3.  Use the :ref:`validate() <config-schema_object-validate>` function to
+    validate configuration values against the schema. In case of a role, call this
+    function inside the role's :ref:`validate() <roles_create_custom_role_validate>` function:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_scalar/http_api.lua
+        :language: lua
+        :start-at: local function validate
+        :end-before: local function apply
+        :dedent:
+
+4.  Obtain values of configuration options using :ref:`get() <config-schema_object-get>`.
+    In case of a role, call it inside the role's :ref:`apply() <roles_create_custom_role_apply>` function:
+
+    ..  literalinclude:: /code_snippets/snippets/config/instances.enabled/config_schema_nodes_scalar/http_api.lua
+        :language: lua
+        :start-at: local function apply
+        :end-before: local function stop
+        :dedent:
+
+
+.. _config_utils_schema_definition:
+
+Defining a schema
+-----------------
+
+A *configuration schema* stores information about a user-defined configuration structure
+that can be passed inside an :ref:`app.cfg <configuration_reference_app_cfg>`
+or a :ref:`roles_cfg <configuration_reference_roles_cfg>` section. It includes
+option names, types, hierarchy, and other aspects of a configuration.
+
+To create a schema, use the :ref:`schema.new() <config-utils-schema-new>` function.
+It has the following arguments:
+
+-   schema name -- an arbitrary string to use as an identifier
+-   root schema node -- a table describing the hierarchical schema structure
+    starting from the root.
+-   (optional) methods -- functions that can be called on this schema object.
```

Comment:
I would say 'user-defined methods' here to highlight that it is not some kind of filter of the built-in methods.

BTW, I would say 'method' instead of 'function' for functions that take the schema object as the first argument and can be called using a colon: `<schema object>:foo(<...>)`.

Reply:
Thanks. 
- Added `user-defined` to this line (methods definition)
- Replaced `function` with `method` when referring to schema object's methods across this page.